net-core/Ical.Net/Ical.Net/DataTypes/WeekDay.cs

```diff
@@ -64,6 +66,20 @@ public override void CopyFrom(ICopyable obj)
             }
         }
 
+        public int CompareTo(WeekDay other)
+        {
+            if (other == null)
```

Comment:
Throwing an exception in a `CompareTo` method probably isn't a good idea. Consider the usage in a LINQ query:

```csharp
var orderedDays = weekdayCollection
    .OrderBy(w => w)
    .ToList();
```

That could `throw` in the event that WeekDay is null. I think being able to sort `null` is reasonable behavior. Consider instead:

```csharp
public int CompareTo(WeekDay other)
{
    if (ReferenceEquals(this, other)) return 0;
    return ReferenceEquals(null, other)
        ? 1
        : DayOfWeek.CompareTo(other.DayOfWeek);
}
```